### PR TITLE
r/lb_listner_certificate - import support

### LINKED ISSUE #13610: Resource aws_lb_listener_certificate doesn't support import
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently we are unable to `import `resource `aws_lb_listener_certificate`.
Error received:
```
resource aws_lb_listener_certificate doesn't support import
````
I would like to request support be added for `import` to support this resource.

### New or Affected Resource(s)

Possibly
* aws_lb_listener
* aws_lb
* aws_alb_listener
* aws_alb
* aws_alb_listener_certificate

### Potential Terraform Configuration

n/a

```hcl
data "aws_acm_certificate" "example_com" {
    domain      = "example.com"
    statuses    = ["ISSUED"]
    most_recent = "true"
}
resource "aws_lb_listener_certificate" "example_com" {
    listener_arn    = "##SOME_ARN_STRING##"
    certificate_arn = "${data.aws_acm_certificate.example_com.arn}"
}
```

### References

* https://www.terraform.io/docs/providers/aws/r/lb_listener_certificate.html
* https://docs.aws.amazon.com/cli/latest/reference/elbv2/describe-listener-certificates.html

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #13610

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_lb_listner_certificate - import support
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAwsLbListenerCertificate_'
--- PASS: TestAccAwsLbListenerCertificate_basic (266.40s)
--- PASS: TestAccAwsLbListenerCertificate_multiple (450.33s)
--- PASS: TestAccAwsLbListenerCertificate_disappears (265.82s)
```

### CODE REVIEW

File: aws/internal/service/elbv2/id.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package elbv2
+
+import (
+	"fmt"
+	"strings"
+)
+
+const listnerCertificateIDSeparator = "_"
```
Comment (ewbankkit) [inline aws/internal/service/elbv2/id.go:8]:
```suggestion
const listenerCertificateIDSeparator = "_"
```

File: aws/internal/service/elbv2/id.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+func ListnerCertificateParseID(id string) (string, string, error) {
```
Comment (ewbankkit) [inline aws/internal/service/elbv2/id.go:10]:
```suggestion
func ListenerCertificateParseID(id string) (string, string, error) {
```

File: aws/internal/service/elbv2/id.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	parts := strings.Split(id, listnerCertificateIDSeparator)
```
Comment (ewbankkit) [inline aws/internal/service/elbv2/id.go:11]:
```suggestion
	parts := strings.Split(id, listenerCertificateIDSeparator)
```

File: aws/internal/service/elbv2/id.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	if len(parts) == 2 && parts[0] != "" && parts[1] != "" {
+		return parts[0], parts[1], nil
+	}
+
+	return "", "",
+		fmt.Errorf("unexpected format for ID (%q), expected listner-arn"+listnerCertificateIDSeparator+
```
Comment (ewbankkit) [inline aws/internal/service/elbv2/id.go:17]:
```suggestion
		fmt.Errorf("unexpected format for ID (%q), expected listner-arn"+listenerCertificateIDSeparator+
```

File: aws/resource_aws_lb_listener_certificate.go [hunk 5]
```diff
@@ -35,16 +41,19 @@ func resourceAwsLbListenerCertificate() *schema.Resource {
 func resourceAwsLbListenerCertificateCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).elbv2conn
 
+	listnerArn := d.Get("listener_arn").(string)
```
Comment (ewbankkit) [inline aws/resource_aws_lb_listener_certificate.go:44]:
```suggestion
	listenerArn := d.Get("listener_arn").(string)
```

File: aws/resource_aws_lb_listener_certificate.go [hunk 6]
```diff
@@ -35,16 +41,19 @@ func resourceAwsLbListenerCertificate() *schema.Resource {
...
+	certificateArn := d.Get("certificate_arn").(string)
+
 	params := &elbv2.AddListenerCertificatesInput{
-		ListenerArn: aws.String(d.Get("listener_arn").(string)),
+		ListenerArn: aws.String(listnerArn),
```
Comment (ewbankkit) [inline aws/resource_aws_lb_listener_certificate.go:48]:
```suggestion
		ListenerArn: aws.String(listenerArn),
```

File: aws/resource_aws_lb_listener_certificate.go [hunk 7]
```diff
@@ -35,16 +41,19 @@ func resourceAwsLbListenerCertificate() *schema.Resource {
...
 		Certificates: []*elbv2.Certificate{
 			{
-				CertificateArn: aws.String(d.Get("certificate_arn").(string)),
+				CertificateArn: aws.String(certificateArn),
 			},
 		},
 	}
 
-	log.Printf("[DEBUG] Adding certificate: %s of listener: %s", d.Get("certificate_arn").(string), d.Get("listener_arn").(string))
+	log.Printf("[DEBUG] Adding certificate: %s of listener: %s", certificateArn, listnerArn)
```
Comment (ewbankkit) [inline aws/resource_aws_lb_listener_certificate.go:56]:
```suggestion
	log.Printf("[DEBUG] Adding certificate: %s of listener: %s", certificateArn, listenerArn)
```

File: aws/resource_aws_lb_listener_certificate.go [hunk 8]
```diff
@@ -66,24 +75,26 @@ func resourceAwsLbListenerCertificateCreate(d *schema.ResourceData, meta interfa
 	}
 
 	if err != nil {
-		return fmt.Errorf("error adding LB Listener Certificate: %s", err)
+		return fmt.Errorf("error adding LB Listener Certificate: %w", err)
 	}
 
-	d.SetId(d.Get("listener_arn").(string) + "_" + d.Get("certificate_arn").(string))
+	d.SetId(fmt.Sprintf("%s_%s", listnerArn, certificateArn))
```
Comment (ewbankkit) [inline aws/resource_aws_lb_listener_certificate.go:81]:
```suggestion
	d.SetId(tfelbv2.ListenerCertificateCreateID(listenerArn, certificateArn))
```

and add `ListenerCertificateCreateID()`.

File: aws/resource_aws_lb_listener_certificate.go [hunk 9]
```diff
@@ -66,24 +75,26 @@ func resourceAwsLbListenerCertificateCreate(d *schema.ResourceData, meta interfa
...
 
 	return resourceAwsLbListenerCertificateRead(d, meta)
 }
 
 func resourceAwsLbListenerCertificateRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).elbv2conn
 
-	certificateArn := d.Get("certificate_arn").(string)
-	listenerArn := d.Get("listener_arn").(string)
+	listenerArn, certificateArn, err := tfelbv2.ListnerCertificateParseID(d.Id())
```
Comment (ewbankkit) [inline aws/resource_aws_lb_listener_certificate.go:89]:
```suggestion
	listenerArn, certificateArn, err := tfelbv2.ListenerCertificateParseID(d.Id())
```

File: aws/resource_aws_lb_listener_certificate.go [hunk 10]
```diff
@@ -66,24 +75,26 @@ func resourceAwsLbListenerCertificateCreate(d *schema.ResourceData, meta interfa
...
 }
 
 func resourceAwsLbListenerCertificateRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).elbv2conn
 
-	certificateArn := d.Get("certificate_arn").(string)
-	listenerArn := d.Get("listener_arn").(string)
+	listenerArn, certificateArn, err := tfelbv2.ListenerCertificateParseID(d.Id())
+	if err != nil {
+		return err
```
Comment (bflad) [inline aws/resource_aws_lb_listener_certificate.go:91]:
```suggestion
		return fmt.Errorf("error parsing ELBv2 Listener Certificate ID (%s): %w", d.Id(), err)
```

**Review by bflad — COMMENTED** [review]
I think just this and changelog 👍

**Comment by bflad** [comment]
Oh yes, we need to add `ImportState` testing as well. 👍

**Comment by DrFaust92** [comment]
Added Import tests + changelog

File: .changelog/16474.txt [hunk 11]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+resource/aws_lb_listner_certificate: Add import support
```
Comment (bflad) [inline .changelog/16474.txt:2]:
```suggestion
resource/aws_lb_listener_certificate: Add import support
```

Comment (DrFaust92) [reply .changelog/16474.txt]: Doh

**Review by bflad — APPROVED** [review]
Looks great, thank you, @DrFaust92 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAwsLbListenerCertificate_disappears (170.83s)
--- PASS: TestAccAwsLbListenerCertificate_basic (176.25s)
--- PASS: TestAccAwsLbListenerCertificate_multiple (234.15s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAwsLbListenerCertificate_disappears (155.87s)
--- PASS: TestAccAwsLbListenerCertificate_basic (159.04s)
--- PASS: TestAccAwsLbListenerCertificate_multiple (251.86s)
```

**Comment by greedy52** [comment]
`ListenerCertificateParseID` is a breaking change in 3.28.0 when the certificate name (ex: iam cert) has `_` in it, as `Split` will return more than 2 tokens.

**Pull request merged**